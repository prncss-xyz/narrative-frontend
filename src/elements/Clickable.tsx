import { Box } from "./basics";

export function Clickable({
  onClick,
  children,
  disabled,
  ...props
}: {
  onClick: () => void;
  children: React.ReactNode;
  disabled?: boolean;
  [prop: string]: unknown; // TODO: could be more restrictive
}) {
  return (
    <Box>
      <Box
        as="button"
        backgroundColor="inherit"
        borderStyle="none"
        padding="0px"
        height="100%"
        cursor={disabled ? undefined : "pointer"}
        disabled={disabled}
        onClick={onClick}
        {...props}
      >
        {children}
      </Box>
    </Box>
  );
}
