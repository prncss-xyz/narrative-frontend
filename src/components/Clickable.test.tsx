import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Clickable } from "./Clickable";

describe("Clickable", () => {
  describe("disabled", () => {
    it("should trigger when clicked", async () => {
      const spy = vi.fn();
      const { container } = render(
        <Clickable onClick={spy}>not disabled</Clickable>
      );
      await userEvent.click(screen.getByText("not disabled"));
      expect(spy).toHaveBeenCalledTimes(1);
      expect(container).toMatchSnapshot();
    });
    it("should trigger when clicked", async () => {
      const spy = vi.fn();
      const { container } = render(
        <Clickable disabled={true} onClick={spy}>
          disabled
        </Clickable>
      );
      await userEvent.click(screen.getByText("disabled"));
      expect(spy).toHaveBeenCalledTimes(0);
      expect(container).toMatchSnapshot();
    });
  });
});
