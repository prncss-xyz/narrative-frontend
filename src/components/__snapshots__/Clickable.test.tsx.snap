// Vitest Snapshot v1

exports[`Clickable > disabled > should trigger when clicked 1`] = `
<div>
  <div
    class="css-1x3sxzl-Box e2i8k6n3"
  >
    <button
      class="css-19fpo8m-Box e2i8k6n3"
      cursor="pointer"
      height="100%"
    >
      not disabled
    </button>
  </div>
</div>
`;

exports[`Clickable > disabled > should trigger when clicked 2`] = `
<div>
  <div
    class="css-1x3sxzl-Box e2i8k6n3"
  >
    <button
      class="css-84vwd6-Box e2i8k6n3"
      disabled=""
      height="100%"
    >
      disabled
    </button>
  </div>
</div>
`;
